File: NewsApp/app/src/main/java/com/example/shakil/newsapp/Adapter/ListSourceAdapter.java
package com.example.shakil.newsapp.Adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.shakil.newsapp.Model.WebSite;
import com.example.shakil.newsapp.R;

import de.hdodenhof.circleimageview.CircleImageView;

public class ListSourceAdapter extends RecyclerView.Adapter<ListSourceAdapter.ListSourceViewHolder> {

    private Context context;
    private WebSite webSite;

    public ListSourceAdapter(Context context, WebSite webSite) {
        this.context = context;
        this.webSite = webSite;
    }

    @NonNull
    @Override
    public ListSourceViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ListSourceViewHolder(LayoutInflater.from(context).inflate(R.layout.source_layout, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull ListSourceViewHolder holder, int position) {
        holder.source_title.setText(webSite.getSources().get(position).getName());
    }

    @Override
    public int getItemCount() {
        return webSite.getSources().size();
    }

    public class ListSourceViewHolder extends RecyclerView.ViewHolder {

        TextView source_title;
        CircleImageView source_image;

        public ListSourceViewHolder(@NonNull View itemView) {
            super(itemView);

            source_title = itemView.findViewById(R.id.source_name);
            source_image = itemView.findViewById(R.id.source_image);
        }
    }
}

File: KotlinNewsApp/app/src/main/java/com/example/shakil/kotlinnewsapp/Adapter/ListSourceAdapter.kt
package com.example.shakil.kotlinnewsapp.Adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.shakil.kotlinnewsapp.Interface.ItemClickListener
import com.example.shakil.kotlinnewsapp.Model.WebSite
import com.example.shakil.kotlinnewsapp.R
import kotlinx.android.synthetic.main.source_news_layout.view.*

class ListSourceAdapter(val context: Context, val webSite: WebSite): RecyclerView.Adapter<ListSourceAdapter.ListSourceViewHolder>() {

    class ListSourceViewHolder(itemView: View): RecyclerView.ViewHolder(itemView), View.OnClickListener {
        private lateinit var itemClickListener: ItemClickListener

        var source_title = itemView.source_news_name
        init {
            itemView.setOnClickListener(this)
        }

        fun setItemClickListener(itemClickListener: ItemClickListener) {
            this.itemClickListener = itemClickListener
        }

        override fun onClick(v: View?) {
            itemClickListener.onClick(v!!, adapterPosition)
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ListSourceViewHolder {
        val inflater = LayoutInflater.from(context)
        val itemView = inflater.inflate(R.layout.source_news_layout, parent, false)
        return ListSourceViewHolder(itemView)
    }

    override fun getItemCount(): Int {
        return webSite.sources!!.size
    }

    override fun onBindViewHolder(holder: ListSourceViewHolder, position: Int) {
        holder.source_title.text = webSite.sources!![position].name

        holder.setItemClickListener(object : ItemClickListener {
            override fun onClick(view: View, position: Int) {
                TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
            }
        })
    }
}